bootstrapper/internal/journald/journald_test.go

```diff
@@ -7,58 +7,131 @@ SPDX-License-Identifier: AGPL-3.0-only
 package journald
 
 import (
+	"bytes"
 	"errors"
+	"io"
 	"os/exec"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-type stubJournaldCommand struct {
-	outputReturn []byte
-	outputError  error
+type stubCommand struct {
+	startCalled bool
+	startErr    error
+	waitErr     error
 }
 
-func (j *stubJournaldCommand) Output() ([]byte, error) {
-	return j.outputReturn, j.outputError
+func (j *stubCommand) Start() error {
+	j.startCalled = true
+	return j.startErr
 }
 
-func TestCollect(t *testing.T) {
+func (j *stubCommand) Wait() error {
+	return j.waitErr
+}
+
+type stubReadCloser struct {
+	reader   io.Reader
+	readErr  error
+	closeErr error
+}
+
+func (s *stubReadCloser) Read(p []byte) (n int, err error) {
+	if s.readErr != nil {
+		return 0, s.readErr
+	}
+	return s.reader.Read(p)
+}
+
+func (s *stubReadCloser) Close() error {
+	return s.closeErr
+}
+
+func TestPipe(t *testing.T) {
```

Comment:
While there is no rule that tests must come first in a `_test.go` file, the general idea is to habe the most important things on top. Generally, we came to the conclusion that the `TestXxx()` functions are more important than the stub implementations. Therefore, please change the order accordingly.